### ISSUE TITLE
Measures controleren (na bevestiging nieuwe parameters)

### ISSUE THREAD

**Issue by AuguB** [amarquand/PCNtoolkit#259]

**Comment by ymkeverduyn** [comment]
Hi, 
I reran with the new parameters. The sampling is more stable (i.e. no overflow warnings anymore) but some metrics remain unplausible. 
Especially MSLL and NLL sometimes have extreme large values (see screenshot).

![Image](https://github.com/user-attachments/assets/a4859d03-2cb3-4e87-9caf-9b00ddbada90)

The corresponding plots seem okey. For example alpha_1: 

![Image](https://github.com/user-attachments/assets/9a4ecdb0-d819-4022-8222-18f34ac2480e)

Thank you for taking a look at it and let me know if you need other information.

**Comment by AuguB** [comment]
Good to hear that the reparameterization has stabilized the sampler. 

I will have to look into the implementation of the NLL again, I will try to make time for it soon. Could you in the meantime provide the centiles plots for all the measures, for both train and test? You can share those via Slack or email if you want.

**Comment by ymkeverduyn** [comment]
I sent them via email.

**Comment by ymkeverduyn** [comment]
Also , I think the way R2 is calculated is not completely correct. 
I think it should be:
        y = data["Y"].values
        yhat = data["Yhat"].values
        ss_res = np.sum((y - yhat) ** 2)
        ss_tot = np.sum((y - np.mean(y)) ** 2)
        r2 = 1 - ss_res / ss_tot

**Comment by AuguB** [comment]
I had a look at the Alpha runs 0 to 3, and it seems that the mapping of the test set isn’t working properly for some models — see the attachment.

The columns represent the four runs, the top row is the training set, the bottom row is the test set. The two plots with the outliers are exactly the runs with the unlikely metrics. I don’t yet understand why this is happening, and I actually need the corresponding Markov chains (the idata files) for that. If you're able to, could you send me the zipped model folders from these four runs?

I also noticed that you're modeling sex as a batch effect, but there’s only data from a single sex in both your training and test sets. Was this a conscious choice? I predict that removing unnecessary batch effects would further stabilize the models.

About the R2, I think you're right. I will correct it.

**Comment by AuguB** [comment]
In the plots you sent me, the scattered data is called 'control_data_norm_EEG_test', which is different from the data you use to plot the centiles above; `control_data_EEG_test`. What's the difference between these datasets?

**Comment by AuguB** [comment]
If you have time, could you try making the centile plots of the test set for the first 4 splits, once with hamonization, and once without?

**Comment by AuguB** [comment]
You can use this function to make the trace plots:

https://github.com/amarquand/PCNtoolkit/blob/40468883009f7320d800aee6201606a66ca16c66/pcntoolkit/normative_model.py#L428
